### PR TITLE
ADD: devnet

### PR DESCRIPTION
**Pull request by gbayasgalan**
fixes: https://github.com/stereum-dev/ethereum-node/issues/1729
fixes: https://github.com/stereum-dev/ethereum-node/issues/2050

### CODE REVIEW

**Review by MaxTheGeeek — DISMISSED** [review]
lgtm

File: launcher/src/backend/ethereum-services/GethService.js [hunk 1]
```diff
@@ -7,40 +7,71 @@ export class GethService extends NodeService {
...
+        : [
+            `--${network}`,
+            `--datadir=${dataDir}`,
+            "--state.scheme=path",
+            "--http",
+            "--http.port=8545",
+            "--http.addr=0.0.0.0",
+            "--http.vhosts=*",
+            "--http.api=eth,web3,net",
+            "--http.corsdomain=*",
```
Comment (NeoPlays) [inline launcher/src/backend/ethereum-services/GethService.js:52]: i think you can code this simpler. the configs are almost the same, just remove / add the un/wanted ones

File: launcher/src/backend/ServiceManager.js [hunk 2]
```diff
@@ -917,23 +924,34 @@ export class ServiceManager {
...
-          new ServicePort("127.0.0.1", args.port ? args.port : 3500, 3500, servicePortProtocol.tcp),
-        ];
+        ports =
+          args.network === "devnet"
+            ? [
+                new ServicePort(null, 4000, 4000, servicePortProtocol.tcp),
+                new ServicePort(null, 3500, 3500, servicePortProtocol.tcp),
+                new ServicePort(null, 8080, 8080, servicePortProtocol.tcp),
+                new ServicePort(null, 6060, 6060, servicePortProtocol.tcp),
+                new ServicePort(null, 9090, 9090, servicePortProtocol.tcp),
```
Comment (NeoPlays) [inline launcher/src/backend/ServiceManager.js:934]: just push the additional ports if it's devnet. saves code ;)

File: launcher/src/backend/ethereum-services/PrysmBeaconService.js [hunk 3]
```diff
@@ -35,27 +56,52 @@ export class PrysmBeaconService extends NodeService {
...
+            `--genesis-state=${genesisDir}/genesis.ssz`,
+            "--interop-eth1data-votes",
+            `--chain-config-file=${configYamlDir}/config.yml`,
+            "--contract-deployment-block=0",
+            `--chain-id=${chainId}`,
+            "--rpc-host=0.0.0.0",
+            "--grpc-gateway-host=0.0.0.0",
+            `--execution-endpoint=${executionEndpoint}`,
+            "--accept-terms-of-use=true",
+            `--jwt-secret=${JWTDir}`,
```
Comment (NeoPlays) [inline launcher/src/backend/ethereum-services/PrysmBeaconService.js:73]: same here

**Review by NeoPlays — CHANGES_REQUESTED** [review]
This was all i could find.
I'm sure i missed something 😅

File: launcher/src/backend/ServiceManager.js [hunk 4]
```diff
@@ -2340,4 +2360,126 @@ export class ServiceManager {
...
+        `Failed to write genesis file: ${error}`
+      );
+      console.error("Error writing genesis file:", error);
+      throw error;
+    } finally {
+      this.nodeConnection.taskManager.otherTasksHandler(ref);
+    }
+  }
+
+  async writeConfigYaml(writeConfigData) {
```
[the remark this answers is not in the document]
Comment (gbayasgalan) [reply launcher/src/backend/ServiceManager.js:2394]: The function name is already unique, but I added "Devnet" for better clarity

File: controls/roles/initiate-devnet-genesis/tasks/create_genesis.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
+---
+- name: Create beacon chain genesis
+  docker_container:
+    name: create-beacon-chain-genesis
+    image: gcr.io/prysmaticlabs/prysm/cmd/prysmctl:v5.0.4
```
[the remark this answers is not in the document]
Comment (gbayasgalan) [reply controls/roles/initiate-devnet-genesis/tasks/create_genesis.yml:5]: This version is specific to the devnet, atm :)

[the remark this answers is not in the document]
Comment (gbayasgalan) [reply controls/roles/initiate-devnet-genesis/tasks/create_genesis.yml:5]: https://github.com/stereum-dev/ethereum-node/issues/2111

**Comment by daGscheid** [comment]
Will be only be possible currently with a specific client combination and version after release for now: https://github.com/stereum-dev/ethereum-node/issues/2111

**Review by NeoPlays — APPROVED** [review]
lgtm so far

**Comment by gbayasgalan** [comment]
> lgtm so far

lets wait meeting tomorrow :)

**Pull request merged**